distributed/tests/test_worker.py

```diff
@@ -507,26 +510,21 @@ async def test_io_loop(s):
         assert w.io_loop is s.loop
 
 
-@gen_cluster(client=True, nthreads=[])
-async def test_spill_to_disk(c, s):
+@gen_cluster(
+    client=True, nthreads=[("", 1)], worker_kwargs={"memory_limit": 1200 / 0.6}
+)
+async def test_spill_to_disk(c, s, w):
     np = pytest.importorskip("numpy")
-    w = await Worker(
-        s.address,
-        loop=s.loop,
-        memory_limit=1200 / 0.6,
-        memory_pause_fraction=None,
-        memory_spill_fraction=None,
-    )
-
-    x = c.submit(np.random.randint, 0, 255, size=500, dtype="u1", key="x")
+    # size on memory 500
+    x = c.submit(np.zeros, 500, dtype="u1", key="x")
     await wait(x)
-    y = c.submit(np.random.randint, 0, 255, size=500, dtype="u1", key="y")
+    y = c.submit(np.zeros, 500, dtype="u1", key="y")
     await wait(y)
```

Comment:
@crusaderky in a previous review you suggested doing `await wait([x, y])` for some reason this consistently causes the priority of `x` and `y` are flipped, and evicting `y` when `z` gets submitted instead of `x`, causing the test to fail. I'm not sure what's the reason why this happens, but that's why I'm keeping each `await wait()` after each submition.

Reply:
if you don't wait for x before you submit y, there's no telling which of the two will land in Worker.data first.